### PR TITLE
feat(storage): Accept `SESSION_TOKEN` for AWS temporary credentials

### LINKED ISSUE #7942: rename SECURITY_TOKEN to SESSION_TOKEN
**Summary**

```
COPY INTO mytable
  FROM 's3://mybucket/data.csv'
  CONNECTION = (
        ENDPOINT_URL = 'https://<endpoint-URL>'
        ACCESS_KEY_ID = '<your-access-key-ID>'
        SECRET_ACCESS_KEY = '<your-secret-access-key>'
        SECURITY_TOKEN = '<your-security-token>'
   )
  FILE_FORMAT = (type = 'CSV' field_delimiter = ','  record_delimiter = '\n' skip_header = 1) size_limit=10;
```

it's hard to search by the key word "aws security token", but easy to get result by keyword "aws session token".

how about change it to SESSION_TOKEN to keep the concept same as the AWS docs?

### PR DESCRIPTION
**Pull request by ClSlaid**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

This PR allows users to offer AWS temporary credentials in the name "SESSION_TOKEN", which is more widely used.

The former names, like "SECURITY_TOKEN" and "AWS_TOKEN", stay available.

Fixes #7942

### CODE REVIEW

**Comment by Xuanwo** [comment]
Please also update docs.

**Review by Xuanwo — APPROVED** [review]

File: src/common/storage/src/location.rs [hunk 1]
```diff
@@ -156,8 +156,9 @@ pub fn parse_uri_location(l: &UriLocation) -> Result<(StorageParams, String)> {
                     .unwrap_or_default(),
                 security_token: l
```
Comment (flaneur2020) [inline src/common/storage/src/location.rs:157]: how about make the variable name into session_token together?

Comment (Xuanwo) [reply src/common/storage/src/location.rs:157]: We can't. It has been persisted in meta.

Comment (ClSlaid) [reply src/common/storage/src/location.rs:157]: the internal remains the same, otherwise, protobuf stuff should be dealt with.

**Comment by Xuanwo** [comment]
@mergify update

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**